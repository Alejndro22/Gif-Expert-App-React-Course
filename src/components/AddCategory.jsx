import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = (e) => {
    e.preventDefault();
    const finalInput = inputValue.trim();
    if (finalInput.length <= 1) return;
    // setCategories((categories) => {
    //   return [inputValue, ...categories];
    // });
    onNewCategory(finalInput);
    setInputValue("");
  };

  return (
    <form onSubmit={onSubmit}>
      <input
        type="text"
        placeholder="Busca gifs"
        value={inputValue}
        // Como recibo el event puedo no mandarle nada, pues es el primer valor por defecto
        // onChange={(e) => onInputChange(e)}
        onChange={onInputChange}
      />
    </form>
  );
};

// Debe ser obligatorio que se envie la funcion onNewCategory
AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
