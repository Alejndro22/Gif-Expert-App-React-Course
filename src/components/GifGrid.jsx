import { getGifs } from "../helpers/getGifs";

export const GifGrid = ({ category }) => {
  // No se deberia ejecutar nunca una funcion dentro de un functional component pues se llama con cada refresh, pero
  // por el momento quedara asi
  getGifs(category);

  return (
    <>
      <h3>{category}</h3>
      <p>Hola Mundo</p>
    </>
  );
};
