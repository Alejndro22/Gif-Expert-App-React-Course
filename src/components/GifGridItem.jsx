import PropTypes from "prop-types";

export const GifGridItem = ({ title, url, id }) => {
  return (
    <div className="card">
      <img src={url} alt={title}></img>
      <p>{title}</p>
    </div>
  );
};

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
