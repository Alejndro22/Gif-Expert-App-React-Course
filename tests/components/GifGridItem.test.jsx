import { render, screen } from "@testing-library/react";
import { GifGridItem } from "../../src/components/GifGridItem";

describe("Pruebas en <GifGridItem />", () => {
  const title = "Valorant";
  const url = "https://www.giphy.com/valorant.gif";

  test("debe hacer match con el Snapshot", () => {
    const { container } = render(<GifGridItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("debe mostrar la imagen con el URL y el ALT indicado", () => {
    render(<GifGridItem title={title} url={url} />);
    // screen.debug();
    // expect(screen.getByRole("img").src).toBe(url);
    // expect(screen.getByRole("img").alt).toBe(title);
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("debe mostrar el titulo en el componente (que exista)", () => {
    render(<GifGridItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
