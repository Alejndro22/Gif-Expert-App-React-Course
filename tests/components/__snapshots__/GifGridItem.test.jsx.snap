// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe hacer match con el Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Valorant"
      src="https://www.giphy.com/valorant.gif"
    />
    <p>
      Valorant
    </p>
  </div>
</div>
`;
